### PR TITLE
feature whitespaces

### LINKED ISSUE #117: do not use css to handle white-space
currently we use css to handle white space. we should rather translate the white space to specific tags.

bug because html tidy would change the layout currently.

- https://github.com/andiwand/OpenDocument.core/blob/development/src/Constants.cpp#L68
- https://github.com/andiwand/OpenDocument.core/blob/development/src/odf/OpenDocumentContentTranslator.cpp#L98

### PR DESCRIPTION
**Pull request by andiwand**
fix #117

### CODE REVIEW

File: odf/src/ContentTranslator.cpp [hunk 1]
```diff
@@ -112,15 +112,15 @@ void SpaceTranslator(const tinyxml2::XMLElement &in, std::ostream &out,
   if (count <= 0) {
     return;
   }
+  out << "<span style=\"white-space: pre-wrap;\">";
```
Comment (TomTasche) [inline odf/src/ContentTranslator.cpp:115]: does that happen for EVERY space now? that sounds like it has the potential to blow up the size of HTML files

File: odf/src/ContentTranslator.cpp [hunk 2]
```diff
@@ -112,15 +112,15 @@ void SpaceTranslator(const tinyxml2::XMLElement &in, std::ostream &out,
...
-    // TODO: use "&nbsp;"?
     out << " ";
   }
+  out << "</span>";
 }
 
 void TabTranslator(const tinyxml2::XMLElement &, std::ostream &out, Context &) {
-  // TODO: use "&emsp;"?
-  out << "\t";
+  out << "<span style=\"white-space: pre-wrap;\">&emsp;</span>";
```
Comment (TomTasche) [inline odf/src/ContentTranslator.cpp:123]: why not use a CSS class?

File: odf/src/ContentTranslator.cpp (hunk 1 above)
Comment (andiwand) [reply odf/src/ContentTranslator.cpp:115]: there is a special tag for spaces `<text:s>` which i think exists for cases with more than a single space i think. only those will be treated here.

File: odf/src/ContentTranslator.cpp (hunk 2 above)
Comment (andiwand) [reply odf/src/ContentTranslator.cpp:123]: i could use a css class as well true. but i didn't see any crucial benefits of that.
initially i wanted to use `<pre>` but that introduces a line break. then i was searching for something like this without a line break and only came across `<code>`. i think it has some side effects so i didn't use it.

**Comment by andiwand** [comment]
waiting for #131

**Pull request merged**